src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -675,7 +675,7 @@ private bool TakeStartLine(SocketInput input)
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                 }
 
-                var requestUrlPath = pathBegin.GetString(pathEnd);
+                var requestUrlPath = pathBegin.GetUtf8String(pathEnd);
```

Comment:
Oooh, https://github.com/benaadams/KestrelHttpServer/commit/6769e1e042691475e1e69abc9dc4077c10c6d9d2 was better. Sorry about that.